### PR TITLE
Reference PMT times and Update SRTrigger

### PR DESCRIPTION
**Pull request by gputnam**
Change SRTrigger filling code to be more in line with the rest of CAFMaker. Setup code to reference all PMT times to the same t0.

Depends on https://github.com/SBNSoftware/sbnanaobj/pull/85.

### CODE REVIEW

**Review by jacoblarkin — APPROVED** [review]
Time changes look good. Thanks for making these changes Gray. I'll let Gianluca comment on the trigger change from a vector to a single value. This seems like a good idea to me since that vector should only ever have a single entry, but I don't know if there was a reason to make it a vector.

**Comment by etworcester** [comment]
Gianluca, do we need to be careful about the 4 us offset? I understand that this will be transparent to analyzers if stage0 is processed after some previous PR went in, but that has not been the case for the Run1 data we've looked at so far. We can handle it at the analysis step if needed, but since we're going to the trouble of making the times correct for all analysis variables, should we try to put special handling in here to get that offset right?

**Review by PetrilloAtWork — COMMENTED** [review]
Nothing wrong.
I would like to hear a comment on the change of variable type from `art::InputTag` to `std::string`.

**Comment by PetrilloAtWork** [comment]
@etworcester The issue may be when you process old `Stage1` data files with the new code from this branch.
But there is a load of issues in that anyway. If the data is decoded (`Stage0`) again with the new software, the veto offset will be taken care of.

File: sbncode/CAFMaker/CAFMakerParams.h [hunk 1]
```diff
@@ -263,7 +263,7 @@ namespace caf
       0,
     };
 
-    Atom<art::InputTag> TriggerLabel {
+    Atom<string> TriggerLabel {
```
[the remark this answers is not in the document]
Comment (gputnam) [reply sbncode/CAFMaker/CAFMakerParams.h]: Fixed

File: sbncode/CAFMaker/CAFMakerParams.h [hunk 2]
```diff
@@ -322,6 +322,12 @@ namespace caf
       Comment("How long from the end of a track to save calo-point information. Set to -1 to save nothing"),
       25.
     };
+
+    Atom<bool> ReferencePMTFromTriggerToBeam {
+      Name("ReferencePMTFromTriggerToBeam"),
+      Comment("Whether to switch the reference time of PMT reco from 'trigger' to 'beam spill' time."),
+      true,
```
[the remark this answers is not in the document]
Comment (gputnam) [reply sbncode/CAFMaker/CAFMakerParams.h]: Fixed

File: sbncode/CAFMaker/CAFMaker_module.cc [hunk 3]
```diff
@@ -246,6 +245,8 @@ class CAFMaker : public art::EDProducer {
 
   void InitVolumes(); ///< Initialize volumes from Gemotry service
 
+  void ReferencePMTTimes(StandardRecord &rec, double PMT_refernce_time);
```
[the remark this answers is not in the document]
Comment (gputnam) [reply sbncode/CAFMaker/CAFMaker_module.cc]: Fixed

File: sbncode/CAFMaker/CAFMaker_module.cc [hunk 4]
```diff
@@ -417,6 +418,25 @@ void CAFMaker::BlindEnergyParameters(StandardRecord* brec) {
   }
 }
 
+void CAFMaker::ReferencePMTTimes(StandardRecord &rec, double PMT_refernce_time) {
```
[the remark this answers is not in the document]
Comment (gputnam) [reply sbncode/CAFMaker/CAFMaker_module.cc]: Fixed

**Comment by gputnam** [comment]
I've addressed all of the comments.

**Pull request merged**